src/server/game/Entities/Object/ObjectGuid.h

```diff
@@ -233,18 +277,33 @@ class PackedGuid
         ByteBuffer _packedGuid;
 };
 
-template<HighGuid high>
-class ObjectGuidGenerator
+
+class ObjectGuidGeneratorBase
 {
-    public:
-        explicit ObjectGuidGenerator(uint32 start = 1) : _nextGuid(start) { }
+public:
+    ObjectGuidGeneratorBase(ObjectGuid::LowType start = 1) : _nextGuid(start) { }
 
-        void Set(uint32 val) { _nextGuid = val; }
-        uint32 Generate();
-        uint32 GetNextAfterMaxUsed() const { return _nextGuid; }
+    virtual void Set(uint32 val) { _nextGuid = val; }
+    virtual ObjectGuid::LowType Generate() = 0;
+    ObjectGuid::LowType GetNextAfterMaxUsed() const { return _nextGuid; }
 
-    private:
-        uint32 _nextGuid;
+protected:
+    static void HandleCounterOverflow(HighGuid high);
+    uint64 _nextGuid;
```

Comment:
Well, I was thinking more of an anal retentive level of memory use efficiency. Like I say it's a low level thing not hurting anything but should be changed nonetheless.

I also think a lot more places where uint32/64 are used in 3.3.5 and 6.x should be changed to ObjectGuid::LowType just to make code that little bit more portable. Not sure if you agree though.